astroplan/core.py

```diff
@@ -25,82 +40,106 @@ class Observer(object):
             location.
         """
-        raise NotImplementedError()
 
-    def set_environment(self, pressure, relative_humidity, temperature):
-        """
-        Updates the Observer object with environmental conditions.
+        if name is not None:
+            self.name = name
 
-        <longer description>
-
-        Parameters
-        ----------
-        pressure : `~astropy.units.Quantity`
-            The ambient pressure.
+        # If lat/long given instead of EarthLocation, convert them
+        # to EarthLocation
+        if location is None and (latitude is not None and longitude is not None):
+            accepted_latlon_types = [str, Quantity]
+            if (type(latitude) in accepted_latlon_types and
+                type(longitude) in accepted_latlon_types):
+                latitude = Latitude(latitude, unit=u.degree)
+                longitude = Longitude(longitude, unit=u.degree)
 
-        relative_humidity : `~astropy.units.Quantity`
-            The ambient relative humidity.
+            self.location = EarthLocation.from_geodetic(longitude, latitude,
+                                                        elevation)
 
-        temperature :
-            The ambient temperature.
-        """
-        raise NotImplementedError()
+        elif isinstance(location, EarthLocation):
+            self.location = location
 
-    def get_date(self, input_date_time, timezone=None):
-        """
-        Builds an object containing date and time information.
+        else:
+            raise ValueError('Observatory location must be specified with '
+                    'either (1) latitude and longitude in degrees as '
+                    'accepted by astropy.coordinates.Latitude and '
+                    'astropy.coordinates.Latitude, or (2) as an instance of '
+                    'astropy.coordinates.EarthLocation.')
 
-        Default time zone is UTC.
-        If time zone (local or otherwise) requested, then uses `datetime.datetime`
-        and pytz.timezone to convert from UTC.
+        # Accept various timezone inputs, default to UTC
+        if isinstance(timezone, datetime.tzinfo):
+            self.timezone = timezone
+        elif isinstance(timezone, str) or isinstance(timezone, unicode):
+            self.timezone = pytz.timezone(timezone)
+        else:
+            raise ValueError('timezone keyword should be a string, or an '
+                               'instance of datetime.tzinfo')
 
-        Must first create date_time object with get_date() before using any other 
-        Observer methods, all of which require one as an argument.
+    def altaz(self, target, time):
+        """
+        Returns an instance of `~astropy.coordinates.SkyCoord` with altitude and
+        azimuth of the `FixedTarget` called `target` at time `time`.
 
         Parameters
         ----------
-        input_date_time : WHAT TYPE IS astropy.time OBJECT?
+        target : `~astroplan.FixedObject`
```

Comment:
This is an example of a place where we could get away with accepting either a `Target` *or* a coordinate: you can just do something like:
```
if hasattr(target, 'coord'):
    coord = target.coord
else:
    coord = target
```
and then below replace all `target.coord` references with just `coord`.
(also, I think you meant `FixedTarget` here?)